### PR TITLE
Fixes parse errors for function names starting with the word "operator"

### PR DESCRIPTION
**Pull request by MichaelBaker**
I ran into this issue when trying to bind against Unreal Engine's headers. I don't know if you consider this a valid fix and I couldn't get the bindgen tests to run. In any event, I thought it might be useful for you to see how I fixed the problem for my project.

Prior to this commit, the parser would fail on function names with a
prefix of "operator", event if they weren't actual operators. Here's
the definition that uncovered the error:

```
@property (nullable, readonly, copy) NSArray<NSNumber *> *operators;
```

This commit adds a regular expression that more accurately captures
Objective C's identifier syntax.

### CODE REVIEW

File: src/ir/function.rs [hunk 1]
```diff
@@ -327,6 +328,11 @@ impl FunctionSig {
         ctx: &mut BindgenContext,
     ) -> Result<Self, ParseError> {
         use clang_sys::*;
+
+        lazy_static! {
+            static ref OperatorRegex: Regex = Regex::new("^operator[^a-zA-Z0-9_]").unwrap();
```
Comment (emilio) [inline src/ir/function.rs]: nit: `OPERATOR_REGEX`.

File: src/ir/function.rs [hunk 2]
```diff
@@ -336,7 +342,7 @@ impl FunctionSig {
 
         // Don't parse operatorxx functions in C++
         let spelling = cursor.spelling();
-        if spelling.starts_with("operator") {
+        if OperatorRegex.is_match(spelling.as_str()) {
```
Comment (emilio) [inline src/ir/function.rs]: nit: `is_match(&spelling)`?

**Comment by MichaelBaker** [comment]
I'll see if I can get the tests and stuff working this week some time.

**Comment by emilio** [comment]
I guess there's a much easier way to test this which doesn't involve objective C, I can add the test myself if you want to.

**Comment by kulp** [comment]
In the interest of cleaning up this old pull request, I pushed a new PR #1817 that might be able to close this PR.

**Comment by kulp** [comment]
I am closing the current PR in favor of #1817.

**Pull request closed without merging**